cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -435,31 +435,14 @@ public List<ApplicationId> getAllAppVersionsAppIds(ApplicationReference appRef)
    */
   public Map<ApplicationId, ApplicationMeta> getApplicationsForAppIds(Collection<ApplicationId> appIds)
     throws IOException {
-    List<Range> multiRanges = new ArrayList<>();
-
-    // multiScan for all apps
-    for (ApplicationId appId: appIds) {
-      // For -SNAPSHOT version, ignore the version in the key and fetch with latest = true instead
```

Comment:
This multiScan is still wrong for pre 6.8 pipelines, because `latest` is NULL and we're not able to fetch it in a multiScan or mutiRead fashion